crates/sui-indexer/src/handlers/objects_snapshot_processor.rs

```diff
@@ -138,14 +161,22 @@ where
             }
         }
 
-        info!("Objects snapshot processor starts updating objects_snapshot periodically...");
-        loop {
+        // We are not in backfill mode but it's possible that the snapshot checkpoint is behind the
```

Comment:
Wouldn't the snapshot checkpoint always be behind though

Reply:
Ah I see, this is to handle the scenario where we've started populating the buffer, but the lhs of the buffer is larger than the max checkpoint of snapshot. In that case, we must still upsert from objects history

I think we can speed up this process in a similar vein by first fetching the object updates, and then separately doing the upsert? Instead of relying on the current upsert logic which takes around 20 mins to index 600 checkpoints

Reply:
it't possible to be faster, but reading those object updates and upsert has the networking overhead of passing the data back-n-forth.